gapis/api/gles/compat.go

```diff
@@ -1289,6 +1296,48 @@ func canUsePrecompiledShader(c *Context, d *device.OpenGLDriver) bool {
 	return c.Constants.Vendor == d.Vendor && c.Constants.Version == d.Version
 }
 
+// disableUnusedAttribArrays disables all vertex attribute arrays that are not
+// used by the currently bound program. This is a compatibility fix for devices
+// that will error with GL_INVALID_OPERATION if there's an enabled (but unused)
+// vertex attribute array that has no array data when drawing. AFAICT, this
+// particular behavior is undefined according to the spec.
+func disableUnusedAttribArrays(
+	ctx context.Context,
+	c *Context,
+	dID api.CmdID,
+	cb *CommandBuilder,
+	out transform.Writer) func() {
+
+	p := c.Bound.Program
+	if p == nil || p.ActiveResources == nil {
+		return nil
+	}
+	inputs := p.ActiveResources.ProgramInputs
+	used := make([]bool, c.Constants.MaxVertexAttribBindings)
+	for _, input := range inputs.Range() {
+		for _, l := range input.Locations.Range() {
+			if l < GLint(len(used)) {
```

Comment:
GL can sometimes return locations which are -1, ignore those in the condition? (or make the array into a map)

Reply:
Done